server.js

```diff
@@ -22,23 +22,45 @@ io.on('connection', (socket) => {
     console.log('Client Connected')
 
     socket.on('chat', (message) => {
-        const room = Object.keys(socket.rooms)
-            .filter(x => x !== socket.id)
-
-        io.to(room[0]).emit('chat', message)
+        const room = filterRooms(socket)
+        notifyRoom(room, 'chat', message)
         console.log(message)
     })
 
     socket.on('private', (sender, message) => {
         // private message received
     })
-
-    socket.on('disconnect', () => {
-        io.emit('client disconnected')
+    
+    socket.on('leave-room', (name) => {
+        const room = filterRooms(socket)
+        const message = name + ' has disconnected'
+        notifyRoom(room, 'leave-room', message)
+        console.log(message + ' from ' + room)
+        
+        socket.leave(room, (err) => {
+            if (err) {
+                console.log(err)
+            }
+        })
     })
 
-    socket.on('create', (room) => {
+    socket.on('join-room', (room, name) => {
         socket.join(room)
+        const message = name + ' has connected'
+        notifyRoom(room, 'join-room', message)
+        console.log(message + ' to ' + room)
     })
 })
 
+function notifyRoom(room, event, message) {
+    if (typeof(room) !== 'string') {
+        // if the socket is passed
+        room = filterRooms(room)
+    }
```

Comment:
Is there a reason why we can't filter for the room in `socket.on('chat',` then pass it into notify room? Just bc I feel like it would be a cleaner approach